contracts/test/DaiEchidnaTest.sol

```diff
@@ -24,14 +26,18 @@ contract DaiEchidnaTest {
 
     // --- Math ---
     function add(uint256 x, uint256 y) internal pure returns (uint256 z) {
-        require((z = x + y) >= x);
+        z = x + y;
+        assert (z >= x); // check if there is an addition overflow
     }
     function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
-        require((z = x - y) <= x);
+        z = x - y;
+        assert (z <= x); // check if there is a subtraction overflow
     }
 
     /// @dev Test that supply and balance hold on mint
     function mint(uint256 wad) public {
+        wad = wad % uint256(-1);
```

Comment:
You can wind up causing overflows this way. I'd define a constant `MAX_SUPPLY` and then set `wad = wad % (MAX_SUPPLY - dai.totalSupply());`. The value of `MAX_SUPPLY` should be some reasonable upper bound on what we expect DAI supply could reach. Maybe `10**15 * WAD`, for example.

Reply:
Done.